File: cpp/problem206.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    ListNode* reverseList(ListNode* head) {
        ListNode *p = nullptr, *q = head, *r = nullptr;
        while(q){
            r = q->next;
            q->next = p;
            p = q;
            q = r;
        }
        return p;
    }
};

File: python/problem206.py
class Solution:
    def reverseList(self, head):
        """
        :type head: ListNode
        :rtype: ListNode
        """
        queue=[]
        node=head
        while node:
            queue.append(node.val)
            node=node.next
        node=head
        while queue:
            node.val=queue.pop()
            node=node.next
        return head
        